jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSQueueMessageConsumer.java

```
@@ -0,0 +1,121 @@
+        this.destName = queue;
+        connectionFactory = (ConnectionFactory) context.lookup(connectionFactoryName);
+        connection = connectionFactory.createConnection(username, password);
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        destination = session.createQueue(queue);
+        consumer = session.createConsumer(destination);
+        connection.start();
+    }
+    
+    /**
+     * Called by the JMS Class, and used to read the most recent JMS Message from the associated queue
+     * @return A map containing the message, as well as the queue name and the JMS Message Type
+     * @throws JMSException
+     */
+    public Map<String, Object> consumeMessage() throws JMSException {
+        Message message = consumer.receive(1000);
+        return formatMessage(message);
+    }
+    
+    /**
+     * A helper function to consumeMessage(), used to parse the incoming message and format it according to its message type
+     * @param message           The retrieved message from the queue
+     * @return                  A map containing the message, as well as the queue name and the JMS Message Type
+     * @throws JMSException
+     */
+    public Map<String, Object> formatMessage(Message message) throws JMSException {
+        Map<String, Object> msgMap = new LinkedHashMap<String, Object>();
+        
+        if (message instanceof TextMessage) {
+            String msgText = ((TextMessage) message).getText();
+            msgMap.put("message", msgText);
+            msgMap.put("JMSFormat", TEXT);
+            msgMap.put("destination", this.destName);
+        } else if (message instanceof ObjectMessage) {
+            Object msgObject = ((ObjectMessage) message).getObject();
+            msgMap.put("message", msgObject);
+            msgMap.put("JMSFormat", OBJECT);
+            msgMap.put("destination", this.destName);
+        } else if (message instanceof MapMessage) {
+            
```

FIXME's

probably good practice here (and other places with similar content) to throw exception or otherwise fail spectacularly when called with things it cannot do.  FIXME says I'll fix it, but don't want things to slip out that fail silently.  

throw SomeNotYetSupported('not done yet') type of thing...